xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsEventExecutor.java

```diff
@@ -85,12 +85,17 @@ public void initialize()
     {
         this.queue = this.blockingQueueProvider.initBlockingQueue();
         this.consumers = new ArrayList<>();
+        JMXMentionsMBean mbean = new JMXMentions(this.queue, () -> this.consumers.size());
+        this.jmxRegistration.registerMBean(mbean, MBEAN_NAME);
+    }
+
+    @Override
+    public void startThreads()
```

Comment:
what happens if you call this several times? It will recreate again and again the asked number of threads. You should have some flag to specify that it's already started no?